integration-tests/tests/message_routing.rs

```diff
@@ -0,0 +1,155 @@
+use connection_router::state::{CrossChainId, Message};
+use cosmwasm_std::{Addr, HexBinary, Uint128};
+
+use test_utils::{Chain, Protocol, Worker};
+mod test_utils;
+/// Tests that a single message can be routed fully through the protocol. Submits a message to the
+/// gateway, votes on the poll, routes the message to the outgoing gateway, triggers signing at the prover
+/// and signs via multisig. Also tests that rewards are distributed as expected for voting and signing.
+#[test]
+fn basic_message_routing() {
```

Comment:
Despite the fact that we have not adopted cucumber, can we describe the test cases in its language to some degree? `basic_message_routing` just isn't informative at all. Based on the comments, I'd say `a_single_message_can_be_route_through_and_dealt_by_the_prover` or something like that.